### ISSUE TITLE
Save password popup is quite obtrusive

### ISSUE THREAD

**Issue by swiehr** [kee-org/KeeFox#557]
The save password popup is quite annoying - previously as a toolbar it was less obtrusive for all the cases where you just wanted to ignore and not save/update (e.g. changed captcha in form, etc.)

BTW: Also keep getting the popup for non-login scenarios like simple search forms (e.g. search in a DokuWiki, etc.)

**Comment by luckyrat** [comment]
Any suggestions?

We're pretty limited in what we can offer in terms of UI here in order that we can fit within Firefox API requirements but if you've got a feasible way to improve things I'm keen to hear it.

**Comment by swiehr** [comment]
What was wrong with the old toolbar approach?

**Comment by Borkason** [comment]
Yeah man, the new one sucks donkey ass. It's terribly obstrusive and horribly annoying! I hate it so much. The old approach was way better. And it keeps asking to save passwords, that are already in the database. What the heck?

It's so bad, I am on the verge on looking for alternatives, after being happy for years.

**Comment by luckyrat** [comment]
RE password generation: There are no extra clicks required. Some are optional if you want to use the new functionality but it definitely behaves exactly the same as earlier versions by default.

**Comment by damienleone** [comment]
My suggestion for this is to add an option that would disable the "Save new password" / "Update password" popup and instead show a new icon on top of the KeeFox toolbar logo.

This way, one would know that it is possible to add / update an entry by just looking at the toolbar.  Once the toolbar is clicked, the popup would be the same as before.  The only difference is that it wouldn't show up by default.

In a nutshell: I do not want any popup to ever show up.  I should manually take action if I want to add / update an entry.

**Comment by marjasin** [comment]
The popups are so god damn intrusive. PLEASE REMOVE THIS SHIT

**Comment by Borkason** [comment]
> In a nutshell: I do not want any popup to ever show up.
:+1: 

Especially not one covering 1/10th of my screen.

**Comment by malesa** [comment]
I agree that displaying this popup is annoying.
On 99% sites I need only to store login and password. And KeeFox constantly asks me to save on sites where password didn't change and I must to close this popup.
Maybe it should have an option for a site (default OFF) to save additional fields. If user didn't change login/password it shouldn't ask anything.

**Comment by luckyrat** [comment]
Note that this issue is only about exploring any possible ways to make the popup less obtrusive when it is displayed. Making it appear less often is a different issue (improved in the next KeeFox version): #570

**Comment by chronometric** [comment]
OK, for making it less obtrusive: 

1. Ignoring the aesthetics of the UI elements for the moment, just un-bloat them a little. The toolbar options menu is twice as tall and almost 3x as wide as it needs to be. I have almost 10 other addons around my browser with toolbar popups, and they all make use of very simple elements like most windows popups, or maybe a simple text list like Session Manager,  instead of trying to look like a browser toolbar-themed box with buttons on it. It clashes with every other aesthetic of modern FF even under windows 7, which contributes to the focus pull.

2. Limit open time drastically. Open it for 3-5 seconds, and if the user just can't make it there, have an control interaction on the toolbar that will pop it up for the previous site. Even when needing to modify every single entry, this kind of mouseover approach would work fine, instead of leaving it open until told otherwise, a practice usually only observed for program or system alerts.

3. Two-stage button. Pop up a very small box with a Save and Update button. Then if one is clicked, bring up the window as it appears now (except in this configuration the tabs would be on the bottom).

**Comment by thx1200** [comment]
I would prefer the RoboForm approach.  By default, it doesn't save.  If I hold down ALT and submit, it asks where I want to save it.

As for UI enhancements alone of the pop-up alone, with potential behavior change excluded, I would argue that a bar across the top of the screen that pushes down page content like IE used to do a long time ago with downloads (as opposed to overlaying content), is less intrusive, but equally well at getting user attention.

Thanks for the great addon.  No rude comment from me, I just want workflow improved on an absolutely essential addon for me that I love.

**Comment by chronometric** [comment]
Hate to reiterate, but I just had to perform a two-factor auth on my paypal account using a SMSed number...and the whole time, the KeeFox window is covering up half of the fields I need, and I have a widescreen monitor.

Why isn't it moveable? Resizeable? I forget now, but was it even closeable? It seems like it would take more effort to create these characteristics than to just make a more standard windows-UI popup. I really appreciate experiments with new UI, but these are very, very obtrusive in almost every instance.

**Comment by luckyrat** [comment]
> It seems like it would take more effort to create these characteristics than to just make a more standard windows-UI popup.

This is the standard Firefox UI (Australis CustomizableUI) - much of it is out of KeeFox's control and some of the bits that are under our control are the way they are in order to work around Firefox bugs. There's lots more information about the history of the development in the issues on GitHub if you're interested to dig deeper.

**Comment by twig** [comment]
Hi there,

I've come across this by googling "do you want keefox to save this multi-page password?" and already have disabled ALL notifications from keefox in the options (even the important ones).

I think that somehow the multi-page notification is ignoring or not adhering to the notification settings.

This was from browsing http://www.rentalcars.com

**Comment by chronometric** [comment]
It looks like the dialog can be closed by pressing the KeeFox toolbar button. This is very good, however not entirely intuitive. An X to close (even a small one) in the upper right would generate less confusion, particularly given how often this this appears at certain times....

**Comment by HairyDude** [comment]
I never use this option, and it annoys me whenever it appears. I want to be able to turn it off forever.
